pytest-server-fixtures/pytest_server_fixtures/base.py

```diff
@@ -300,10 +306,15 @@ def kill(self, retries=5):
                 try:
                     pid = int(pid)
                 except ValueError:
-                    # Can't determine process ID, process shutting down or owned by someone else.
-                    pass
+                    print("Can't determine port, process shutting down or owned by someone else")
                 else:
-                    os.kill(pid, self.kill_signal)
+                    try:
+                        os.kill(pid, self.kill_signal)
+                    except OSError as oe:
+                        if oe.args[0] == 3:  # Process doesn't appear to exist.
```

Comment:
FWIW Python provides a nice way of dealing with errno values:

```python
>>> import os
>>> os.strerror(3)
'No such process'
```

So you could write this as:

```python
import errno
# ...
try:
    os.kill(pid, self.kill_signal)
except OSError as oe:
    if oe.errno == errno.ESRCH:
        print("PID {}: {}".format(pid, os.strerror(oe.errno))
    else:
        raise
```

Admittedly ESRCH isn't the most human-friendly, but it's a little more explicit. The other error code aren't quite as bad: https://docs.python.org/2/library/errno.html